### PR TITLE
ISPN-6236 StateTransferFunctionalTest random failures

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-6236

Remove the JoiningNode helper class

### CODE REVIEW

File: core/src/test/java/org/infinispan/statetransfer/StateTransferFunctionalTest.java [hunk 1]
```diff
@@ -66,7 +65,7 @@ protected void createCacheManagers() throws Throwable {
             .useSynchronization(false)
             .recovery().disable();
       configurationBuilder.clustering().sync().replTimeout(30000);
-      configurationBuilder.clustering().stateTransfer().chunkSize(50);
+      configurationBuilder.clustering().stateTransfer().chunkSize(20);
```
Comment (pruivo) [inline core/src/test/java/org/infinispan/statetransfer/StateTransferFunctionalTest.java]: is the chunk-size the cause of the failures?

Comment (danberindei) [inline core/src/test/java/org/infinispan/statetransfer/StateTransferFunctionalTest.java]: Nope. I just wanted to reduce the delay from 2 seconds to 1 second, and I tried to keep the number of `StateTransferResponseCommand`s roughly the same.

Comment (pruivo) [inline core/src/test/java/org/infinispan/statetransfer/StateTransferFunctionalTest.java]: ok. going to test it :)

**Comment by pruivo** [comment]
integrated! thanks @danberindei !

**Comment by gustavonalle** [comment]
There's a bunch of failures in the query module, apparently because of this change:
http://ci.infinispan.org/viewLog.html?buildTypeId=Infinispan_MasterHotspotJdk8&buildId=36119&guest=1
Is there anything that needs to be tweaked?

**Pull request closed without merging**